newIDE/app/src/Utils/GDevelopServices/User.js

```diff
@@ -48,3 +75,68 @@ export const getUserPublicProfile = (
     .get(`${GDevelopUserApi.baseUrl}/user-public-profile/${id}`)
     .then(response => response.data);
 };
+
+const postBadgeIfLocked = (
+  authenticatedUser: AuthenticatedUser,
+  achievementId: string
+): ?Promise<?Badge> => {
+  const {
+    badges,
+    firebaseUser,
+    getAuthorizationHeader,
+    onBadgesChanged,
+  } = authenticatedUser;
+  if (!badges || !firebaseUser) return null;
+  if (badges.map(badge => badge.achievementId).includes(achievementId)) {
+    return null;
+  }
+  console.log(`posting achievement ${achievementId}`);
+
+  const userId = firebaseUser.uid;
+  return getAuthorizationHeader().then(authorizationHeader =>
```

Comment:
am I right thinking we have no way to protect the endpoint to avoid people from giving themselves the badges at this stage ?

Reply:
Only trivial badges will be created this way, so even if someone hacks the system, they can only create themselves the "first web export", "first event", etc.